packages/cubejs-schema-compiler/src/adapter/ClickHouseQuery.ts

```diff
@@ -168,6 +188,43 @@ export class ClickHouseQuery extends BaseQuery {
     return `${fieldAlias} ${direction}`;
   }
 
+  public getCollation() {
+    const useCollation = getEnv('clickhouseUseCollation', { dataSource: this.dataSource });
+    if (useCollation) {
+      return getEnv('clickhouseSortCollation', { dataSource: this.dataSource });
+    }
+    return null;
+  }
+
+  public override orderBy() {
+    //
+    // ClickHouse orders string by bytes, so we need to use COLLATE 'en' to order by string
+    //
+    if (R.isEmpty(this.order)) {
+      return '';
+    }
+
+    const collation = this.getCollation();
+
+    const orderByString = R.pipe(
+      R.map((order) => {
+        let orderString = this.orderHashToString(order);
+        if (collation && this.getFieldType(order) === 'string') {
+          orderString = `${orderString} COLLATE '${collation}'`;
+        }
+        return orderString;
+      }),
+      R.reject(R.isNil),
+      R.join(', ')
+    )(this.order);
```

Comment:
Don't you mind rewriting this in plain JS instead of using Ramda?
Smth like this:
```suggestion
if (this.order.length === 0) {
  return '';
}

const collation = this.getCollation();

const orderByString = this.order
  .map((order) => {
    let orderString = this.orderHashToString(order);
    if (collation && this.getFieldType(order) === 'string') {
      orderString = `${orderString} COLLATE '${collation}'`;
    }
    return orderString;
  })
  .filter(Boolean) // Analogue `R.reject(R.isNil)`
  .join(', ');
```